### PR TITLE
SDXL Turbo support and example launch

### PR DESCRIPTION
**Pull request by bram-w**
# What does this PR do?

I added an `--is_turbo` argument to the new Diffusion-DPO scripts to support DPO tuning of the 4-step SDXL Turbo model. I also added an example call in the corresponding README

## Before submitting
- [ x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x ] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ x] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
No, but direct follow-up to https://github.com/huggingface/diffusers/pull/6422
- [x ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
Added readme launch and in-line comments
- [ x] Did you write any new necessary tests?
N/A

## Who can review?

@sayakpaul

### CODE REVIEW

File: examples/research_projects/diffusion_dpo/README.md [hunk 1]
```diff
@@ -61,6 +61,34 @@ accelerate launch train_diffusion_dpo_sdxl.py \
   --push_to_hub
 ```
 
+## SDXL Turbo training command
```
Comment (sayakpaul) [inline examples/research_projects/diffusion_dpo/README.md:64]: Possible to maybe add a note that it enables 4-steps sampling?

File: examples/research_projects/diffusion_dpo/train_diffusion_dpo_sdxl.py [hunk 2]
```diff
@@ -423,6 +423,11 @@ def parse_args(input_args=None):
     parser.add_argument(
         "--enable_xformers_memory_efficient_attention", action="store_true", help="Whether or not to use xformers."
     )
+    parser.add_argument(
+        "--is_turbo",
+        action="store_true",
+        help=("Use if tuning SDXL Turbo instead of SDXL"),
```
Comment (sayakpaul) [inline examples/research_projects/diffusion_dpo/train_diffusion_dpo_sdxl.py:429]: nit: if we remove the help statement out of the brackets and remove the comma it probably will go to line above.

File: examples/research_projects/diffusion_dpo/train_diffusion_dpo_sdxl.py [hunk 3]
```diff
@@ -898,6 +936,10 @@ def collate_fn(examples):
                 timesteps = torch.randint(
                     0, noise_scheduler.config.num_train_timesteps, (bsz,), device=latents.device, dtype=torch.long
                 ).repeat(2)
+                if args.is_turbo:
+                    # Learn a 4 timestep schedule
+                    timesteps_0_to_3 = timesteps % 4
+                    timesteps = 250 * timesteps_0_to_3 + 249
```
Comment (sayakpaul) [inline examples/research_projects/diffusion_dpo/train_diffusion_dpo_sdxl.py:942]: Can this be explained in the comments a bit? I know about the 4-step part that was discussed in the ADD paper for training of the student.

**Review by sayakpaul — APPROVED** [review]
Awesome! Thanks much. 

Just some nits.

**Comment by radames** [comment]
hi @bram-w could we also have this setup for SD-turbo version? which is based on SD2.1, thanks

**Comment by sayakpaul** [comment]
@bram-w let me know if you need any help :)

**Review by radames — COMMENTED** [review]
@bram-w testing your script you might also need to change the guidance scale and num steps on the validation step, you could add these lines 

```python
   guidance_scale = 5.0
    num_inference_steps = 25
    if args.is_turbo:
        guidance_scale = 0.0
        num_inference_steps = 4
```

and add the params here 

https://github.com/huggingface/diffusers/blob/dc7f0247874070eb89b826f082a8737b9b83d8a5/examples/research_projects/diffusion_dpo/train_diffusion_dpo_sdxl.py#L120-L124

and here

https://github.com/huggingface/diffusers/blob/dc7f0247874070eb89b826f082a8737b9b83d8a5/examples/research_projects/diffusion_dpo/train_diffusion_dpo_sdxl.py#L144

wdyt @sayakpaul

**Comment by yiyixuxu** [comment]
any update on this PR?

**Comment by sayakpaul** [comment]
@bram-w this PR is already in pretty good state. Let us know if and how we can help to get this merged.

**Comment by sayakpaul** [comment]
@bram-w a gentle ping :)

**Comment by sayakpaul** [comment]
@radames do you want to push the changes you suggested to the script and we can then merge?

**Comment by radames** [comment]
> @radames do you want to push the changes you suggested to the script and we can then merge?

yes happy to do that, do I have permission to send commits here?

**Comment by sayakpaul** [comment]
I think you should be able to since you are a member of the HF org.

**Comment by radames** [comment]
done @sayakpaul

**Review by sayakpaul — APPROVED** [review]
Thanks a bunch!

**Pull request merged**